src/site/_drafts/what-should-you-measure-to-improve-performance/index.md

```diff
@@ -0,0 +1,175 @@
+
+### Puppeteer
+
+Another great option to script testing is via headless Chrome, which can be
+controlled through the Node API
+[Puppeteer](https://github.com/GoogleChrome/puppeteer). The general idea is to
+start the browser through Puppeteer, navigate to the landing page through the
+[goto](https://github.com/GoogleChrome/puppeteer/blob/master/docs/api.md#pagegotourl-options)
+function,
+[inject Javascript](https://github.com/GoogleChrome/puppeteer/blob/master/docs/api.md#pageevaluatepagefunction-args)
+to fill fields or
+[click](https://github.com/GoogleChrome/puppeteer/blob/master/docs/api.md#pageclickselector-options)
+buttons and proceed through the funnel through further
+[goto](https://github.com/GoogleChrome/puppeteer/blob/master/docs/api.md#pagegotourl-options)
+calls as needed. As a metric the duration of the flow can be measured directly,
+but you could also sum up the FCP, FMP or TTI values of the individual loads of
+the flow.
+[Test website performance with Puppeteer](https://michaljanaszek.com/blog/test-website-performance-with-puppeteer)
+provides an overview of how to get performance metrics via Puppeteer. A very
+simplified example node script could look like this:
+
+```
+const puppeteer = require('puppeteer');
+(async () => {  
+  const browser = await puppeteer.launch();  
+  const page = await browser.newPage();  
+  const start = performance.now();  
+  await page.goto('https://www.store.google.com/landingpage');  
+  await page.goto('[https://www.store.google.com/productpage');  
+  await page.click('#buy_btn');  // click the buy button, which triggers overlay
+basket  
+  await page.waitFor('#close_btn');  // wait until basket overlay is shown  
+  await page.goto('https://www.store.google.com/basket');  
+  await page.goto('https://www.store.google.com/checkout');  
+  console.log('Flow took ' + parseInt((performance.now() - start)/1000) + ' seconds');  
+  await browser.close();  
+})();
+```
+
+This script can easily be automated, and even be made part of the build process
+or perf budgets, and be monitored regularly.
+
+## Re-Engagement
+
+Users will return to your site in different time intervals. Depending on time
+passed, the browser may have less of the website's resources cached, needing
+more network requests. This makes it difficult to estimate performance
+differences across repeat visits in lab tests. It is still recommended to keep
+an eye on this though, and a great lab test tool for repeat visits is
+[WebPageTest](https://www.webpagetest.org/), which has a dedicated option for a
+direct repeat visit:
+
+![image](webpagetest_repeat.png)
+
+To get a better feeling for repeat visit performance in the field use your
+analytics package of choice to segment your performance metrics by user type.
+Here is an example of such a report in Google Analytics:
+
+![image](ga_speed_repeat.png)
+
+A report like this will give you page load times for new and returning users as well.
```

Comment:
Feels like maybe there needs to be a bit more of a conclusion.

Reply:
Added a recap section